libcaf_net/caf/net/http/client.test.cpp

```diff
@@ -0,0 +1,542 @@
+      THEN("the output contains the formatted request") {
+        check_eq(transport->output_as_str(), expected);
+      }
+    }
+  }
+  GIVEN("a multi-line HTTP request with a payload") {
+    auto expected = "POST /foo/bar/index.html HTTP/1.1\r\n"
+                    "Content-Length: 13\r\n"
+                    "Content-Type: plain/text\r\n"
+                    "\r\n"
+                    "Hello, world!"sv;
+    WHEN("the client sends the message") {
+      client->begin_header(net::http::method::post, "/foo/bar/index.html"sv);
+      client->add_header_field("Content-Length", "13");
+      client->add_header_field("Content-Type", "plain/text");
+      client->end_header();
+      auto body = "Hello, world!"sv;
+      client->send_payload(as_bytes(make_span(body)));
+      THEN("the output contains the formatted request") {
+        check_eq(transport->output_as_str(), expected);
+      }
+    }
+  }
+  GIVEN("a multi-line HTTP request with a chunked payload") {
+    auto expected = "POST /foo/bar/index.html HTTP/1.1\r\n"
+                    "Content-Type: plain/text\r\n"
+                    "Transfer-Encoding: chunked\r\n"
+                    "\r\n"
+                    "D\r\n"
+                    "Hello, world!\r\n"
+                    "11\r\n"
+                    "Developer Network\r\n"
+                    "0\r\n"
+                    "\r\n"sv;
+    WHEN("the client sends the message") {
+      client->begin_header(net::http::method::post, "/foo/bar/index.html"sv);
+      client->add_header_field("Content-Type", "plain/text");
+      client->add_header_field("Transfer-Encoding", "chunked");
+      client->end_header();
+      auto chunk1 = "Hello, world!"sv;
+      auto chunk2 = "Developer Network"sv;
+      client->send_chunk(as_bytes(make_span(chunk1)));
+      client->send_chunk(as_bytes(make_span(chunk2)));
+      client->send_end_of_chunks();
+      THEN("the output contains the formatted request") {
+        check_eq(transport->output_as_str(), expected);
+      }
+    }
+  }
+}
+
+OUTLINE("Sending all available HTTP methods") {
+  auto app_ptr = app_t::make();
+  auto client_ptr = net::http::client::make(std::move(app_ptr));
+  auto client = client_ptr.get();
+  auto transport = mock_stream_transport::make(std::move(client_ptr));
+  require_eq(transport->start(nullptr), error{});
+  GIVEN("a http request with <method> method") {
+    auto method_name = block_parameters<std::string>();
+    auto expected = detail::format("{} /foo/bar HTTP/1.1\r\n\r\n", method_name);
+    WHEN("the client sends the <enum value> message") {
+      auto method = block_parameters<net::http::method>();
+      client->begin_header(method, "/foo/bar"sv);
+      client->end_header();
+      THEN("the output contains the formatted request") {
+        check_eq(transport->output_as_str(), expected);
+      }
+    }
+  }
+  EXAMPLES = R"(
+    |  method  |            enum value             |
+    | GET      |  caf::net::http::method::get      |
```

Comment:
This works without the fully qualified name now (and is much more readable this way).

```
    | GET      |  get      |
```